src/components/assignment/js/assignment.js

```diff
@@ -0,0 +1,209 @@
+'use strict';
+
+var UsersCollection = require('./users-collection');
+var WorldsCollection = require('./worlds-collection');
+var ProjectModel = require('./project-model');
+var AssignemntRow = require('./assignment-row');
+var env = require('./defaults');
+
+function setEnvironment(options) {
+    env.set(_.omit(options, 'el'));
+}
+
+var Assignment = function (options) {
+    setEnvironment(options);
+    this.initialize(options);
+};
+
+Assignment.prototype = {
+
+    initialize: function (options) {
+        this.el = typeof options.el === 'string' ? $(options.el)[0] : options.el;
+        this.$el = $(this.el);
+        this.$ = _.partialRight($, this.el);
+
+        this.users = new UsersCollection();
+        this.worlds = new WorldsCollection();
+        this.project = new ProjectModel();
+
+        _.bindAll(this, ['render', 'renderTable', 'toggleControlls', 'saveEdit', 'selectAll', 'usassignSelected', '_showUpdating', '_hideUpdating', 'autoAssignAll']);
+
+        this.bindEvents();
+    },
+
+    bindEvents: function () {
+        this.$el.on('update', 'tr', this.saveEdit);
+        this.$el.on('click', 'input:checkbox:not(#select-all)', this.toggleControlls);
+        this.$el.on('click', '#select-all', this.selectAll);
+        this.$el.on('click', '.unassign-user', this.usassignSelected);
+        this.$el.on('click', '.auto-assign-all', this.autoAssignAll);
+    },
+
+    load: function () {
+        var join = _.after(3, function () {
```

Comment:
I think you could just do `$.when.apply(this, [this.worlds.fetch()...]).then(done);`